### ISSUE TITLE
Cannot synchronization on onedrive

### ISSUE THREAD

**Issue by ethan42411** [laurent22/joplin#2406]
I have two PC that was installed Joplin and  logged-in with the same onedrive account to synchronization. But when I upgrade Joplin from 1.0.175 to 1.0.178 or 1.0.179, all notes was not to be sync and display this error on my second PC when I press synchronization.

> Error: Not a valid URL: https://graph.microsoft.com/v1.0/drive/root:/%E6%87%89%E7%94%A8%E7%A8%8B%E5%BC%8F/Joplin 1/.sync?select=name%2Cfile%2Cfolder%2CfileSystemInfo%2CparentReference

**Comment by poordionis** [comment]
I confirm this bug. I don't receive any errors. The synchronisation button from bottom left or from menu or synchronisation keys don't work.

**Comment by wawacorp** [comment]
I have this bug too. Sync button via onedrive does not work. Both on my Fedora machine as well on android.

**Comment by mugoro** [comment]
@wawacorp you should have some settings problem. This bug affects only PC software version (and maybe Linux). On my Android everything works fine.

**Comment by wawacorp** [comment]
@mugoro I have similar problem on my android app as well. 
Changed is sync to File system to sync to my phone storage, and the button works again. 
What I noticed is the sync start failing when on the dektop app, the authentication of Onedrive gives u a localhost link instead the open up login of outlook.com.

**Comment by mugoro** [comment]
The only way to fix this was to take a clean copy of Joplin v1.0.179 (I work with portable versions). Start making the synchronization settings with Onedrive. Start the synchronization (do not enable encryption until you will be asked for). This will work smooth if your last savings are on Onedrive and not locally. Anyway, if you have many local documents (unsynchronized), before installing a clean Joplin, you may export them and afterwards import them back in the new clean Joplin installation.

**Comment by nathancarter** [comment]
I had a similar bug.  It went like this:

1. Install Joplin for the first time.
1. Add two new notes.
1. Set synchronization setting to OneDrive.
1. When first trying to synchronize, authorize it to access my OneDrive, and complete that process.
1. Try to synchronize again and nothing happens.  Repeat using menu item/button/hotkey, nothing has any effect.  Synchronization status screen says nothing synchronized.
1. Set synchronization setting to DropBox instead.
1. When first trying to synchronize, authorize it to access my Dropbox, and complete that process.
1. Try to synchronize and it works fine the first time.

I would prefer to use OneDrive if possible, because I have more space there through my employer, but I'm using DropBox for now until this bug is resolved.

**Comment by graphixillusion** [comment]
I just got a sync error. So i restarted from scratch but jopling delete everything. Now i have all .md encrypted files in the onedrive folder: is there any way to manually decrypt them?

**Comment by alexdevero** [comment]
I am also experiencing issues with syncing on OneDrive. When I try to sync the app it will sync only few dozens of items and then it stops with "Last error: Error".

**Comment by k-svensson** [comment]
I get the same result as alexdevero, i.e. a few notes get synced (anywhere from 3 to 100+) and the operation ends with the same error message. In case it's helpful, here's the the last entry in the log.txt file:

```
2020-03-29 15:38:28: "Operations completed: "
2020-03-29 15:38:28: "createRemote: 11"
2020-03-29 15:38:28: "Total folders: 2"
2020-03-29 15:38:28: "Total notes: 7465"
2020-03-29 15:38:28: "Total resources: 5502"
2020-03-29 15:38:28: "There was some errors:"
2020-03-29 15:38:28: "Error
Code: UnknownError
Header: {"_headers":{"cache-control":["private"],"content-type":["application/json"],"request-id":["59436e1b-7bdf-4231-a27e-cd5707290c2c"],"client-request-id":["59436e1b-7bdf-4231-a27e-cd5707290c2c"],"x-ms-ags-diagnostic":["{\"ServerInfo\":{\"DataCenter\":\"West Europe\",\"Slice\":\"SliceC\",\"Ring\":\"5\",\"ScaleUnit\":\"003\",\"RoleInstance\":\"AGSFE_IN_99\"}}"],"strict-transport-security":["max-age=31536000"],"date":["Sun, 29 Mar 2020 13:38:29 GMT"],"connection":["close"],"content-length":["198"]}}
Request: GET https://graph.microsoft.com/v1.0/drive/root:/Apps/Joplin/d8eaab31f61446e0acdcd6cd9bb5d7d1.md?select=name%2Cfile%2Cfolder%2CfileSystemInfo%2CparentReference {"select":"name,file,folder,fileSystemInfo,parentReference"} null {"headers":{"Authorization":"bearer EwBwA8l6BAAUO9chh8cJscQLmU+LSWpbnr0vmwwAATTnumxrAkOY9vai124Bp/Jufp+m7JqQn3rO91zOn/SySqatjD1YYZnt6j/PpAu+F6IBI97uQrnXWKqmGrEHmBvlxflourHYqUST0yo6VFW8SU4zaDfhgkWsa4gc5QYICLxguE4rO5JgOKGalckuqCQX3RKIJD2q7xxzCfbdvmD0ZpcYl9tB700KPKfLNKmZ/YDMc+V/XCU0ykj0HI84QL/38Z9XlrJUConf3aaGH7iUAdpnbFkGJHa7rwaxLRcqFYGLYynjEGsKKuKxUnH0Eb68gRx2h7SATi33VhI6WhuiK0SwbBID+pRt7+t5t/viSadhqw+Z/hM7WfKsQFeWlx0DZgAACDRIZ/dm7j1UQAKRq+cvNgI4ZFMbgZhd4XoZtzAYU/l8Styew8OGGefuDwRK+uoUT7oatCOSgApL7KT4sQeRoYAdgd4j0hEG0jPcs5LSWdY0KokWSceUpHiFekqBy9znmmbmEiq7N+lpmn1cNzeFEm4fhM3fXR0H2Q8xtGUMjogMwgU/A/mfqijl/4VF5qhHeXEyp0QggocllDdzth454B+YxFMWcIBMg0h+mY9Oy8EZTq1CZYt5xFXPd8EG+GwvlDfp7JNvf7My9TjhBjQKSIM6WQWSO+6IFS6U2515Y52OKDXYFT+7l9zNkC5bKggnv6vr+s8LEmLoFnlzjv1DJ2PFl59LSnK25Sit7ZaU2+8yXas6l7wrDB
Error
    at OneDriveApi.oneDriveErrorResponseToError (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\onedrive-api.js:124:19)
    at OneDriveApi.exec (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\onedrive-api.js:195:24)
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at async OneDriveApi.execJson (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\onedrive-api.js:249:20)
    at async FileApiDriverOneDrive.statRaw_ (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\file-api-driver-onedrive.js:51:11)
    at async FileApiDriverOneDrive.stat (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\file-api-driver-onedrive.js:60:16)
    at async tryAndRepeat (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\file-api.js:37:19)
    at async FileApi.stat (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\file-api.js:162:18)
    at async Synchronizer.start (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\synchronizer.js:366:22)
    at async timeoutCallback (C:\Users\Kjell\AppData\Local\Programs\Joplin\resources\app.asar\lib\registry.js:131:26)"
```

**Comment by goldeelox** [comment]
I've had similar issues syncing with OneDrive, however, I believe my problems are related to using multiple OneDrive accounts simultaneously (work/personal).
Forcing a token refresh works for me.

MacOS (Joplin 1.0.197):
```
sqlite3 $HOME/.config/joplin-desktop/database.sqlite "DELETE FROM settings WHERE key IS 'sync.3.auth';"
```

You should be prompted to re-auth the next time you attempt to synchronize.

**Comment by suuuper** [comment]
Hi, 
I have the same issue. 

I tried with the solution proposed by @goldeelox , but it doesn't work in my case.

`Error: Access denied
    at OneDriveApi.oneDriveErrorResponseToError (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/onedrive-api.js:124:19)
    at OneDriveApi.exec (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/onedrive-api.js:195:24)
    at async OneDriveApi.execJson (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/onedrive-api.js:249:20)
    at async OneDriveApi.appDirectory (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/onedrive-api.js:76:13)
    at async SyncTargetOneDrive.initFileApi (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/SyncTargetOneDrive.js:74:18)
    at async SyncTargetOneDrive.fileApi (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/BaseSyncTarget.js:73:19)
    at async SyncTargetOneDrive.initSynchronizer (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/SyncTargetOneDrive.js:83:38)
    at async SyncTargetOneDrive.synchronizer (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/BaseSyncTarget.js:109:26)
    at async timeoutCallback (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/registry.js:108:19)"`

**Comment by ryanoblak-okta** [comment]
Same issue here, as well. "Synchronize" button does nothing. Deleting the row as described by @goldeelox prompts me to re-auth but it still doesn't work after that.

**Comment by SKILLSHOT-b** [comment]
Received

> Hi,
> I have the same issue.
> 
> I tried with the solution proposed by @goldeelox , but it doesn't work in my case.
> 
> `Error: Access denied at OneDriveApi.oneDriveErrorResponseToError (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/onedrive-api.js:124:19) at OneDriveApi.exec (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/onedrive-api.js:195:24) at async OneDriveApi.execJson (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/onedrive-api.js:249:20) at async OneDriveApi.appDirectory (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/onedrive-api.js:76:13) at async SyncTargetOneDrive.initFileApi (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/SyncTargetOneDrive.js:74:18) at async SyncTargetOneDrive.fileApi (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/BaseSyncTarget.js:73:19) at async SyncTargetOneDrive.initSynchronizer (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/SyncTargetOneDrive.js:83:38) at async SyncTargetOneDrive.synchronizer (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/BaseSyncTarget.js:109:26) at async timeoutCallback (/tmp/.mount_JoplinC0y7aX/resources/app.asar/lib/registry.js:108:19)"`

Received same issue. Ended up using Dropbox instead.

**Comment by suuuper** [comment]
The issue is still present for me

**Comment by mike-lloyd03** [comment]
I'm having this same issue as well. I've authenticated with OneDrive but syncronization doesn't work at all in the desktop app. Mobile app seems to be working fine however.

**Comment by LordHumphrey** [comment]
I get this error too,and my version is > Joplin 1.7.11 (prod, win32).
Here is log in folder ` joplin-desktop`.

`2021-02-04 10:08:11: OneDriveApi: "Got unhandled error:", "undefined", "Not a valid URL: https://graph.microsoft.com/v1.0/drives/b!2V6N5Mnt0UGjOnR4_NG4oxhzgFJeTYNDnJd-pP2yFUpHSq6z-4phTbMGCYN844SK/root:/应用/Joplin/info.json:/content", "Error: Not a valid URL: https://graph.microsoft.com/v1.0/drives/b!2V6N5Mnt0UGjOnR4_NG4oxhzgFJeTYNDnJd-pP2yFUpHSq6z-4phTbMGCYN844SK/root:/应用/Joplin/info.json:/content
Error: Not a valid URL: https://graph.microsoft.com/v1.0/drives/b!2V6N5Mnt0UGjOnR4_NG4oxhzgFJeTYNDnJd-pP2yFUpHSq6z-4phTbMGCYN844SK/root:/应用/Joplin/info.json:/content
    at Object.shim.fetch (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\shim-init-node.js:347:28)
    at OneDriveApi.<anonymous> (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:275:57)
    at Generator.next (<anonymous>)
    at C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:4:12)
    at OneDriveApi.exec (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:226:16)
    at OneDriveApi.<anonymous> (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:383:41)
    at Generator.next (<anonymous>)
    at C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:8:71"`

**Comment by cat-walk** [comment]
> I get this error too,and my version is > Joplin 1.7.11 (prod, win32).
> Here is log in folder ` joplin-desktop`.
> 
> `2021-02-04 10:08:11: OneDriveApi: "Got unhandled error:", "undefined", "Not a valid URL: https://graph.microsoft.com/v1.0/drives/b!2V6N5Mnt0UGjOnR4_NG4oxhzgFJeTYNDnJd-pP2yFUpHSq6z-4phTbMGCYN844SK/root:/应用/Joplin/info.json:/content", "Error: Not a valid URL: https://graph.microsoft.com/v1.0/drives/b!2V6N5Mnt0UGjOnR4_NG4oxhzgFJeTYNDnJd-pP2yFUpHSq6z-4phTbMGCYN844SK/root:/应用/Joplin/info.json:/content Error: Not a valid URL: https://graph.microsoft.com/v1.0/drives/b!2V6N5Mnt0UGjOnR4_NG4oxhzgFJeTYNDnJd-pP2yFUpHSq6z-4phTbMGCYN844SK/root:/应用/Joplin/info.json:/content at Object.shim.fetch (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\shim-init-node.js:347:28) at OneDriveApi.<anonymous> (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:275:57) at Generator.next (<anonymous>) at C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:8:71 at new Promise (<anonymous>) at __awaiter (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:4:12) at OneDriveApi.exec (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:226:16) at OneDriveApi.<anonymous> (C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:383:41) at Generator.next (<anonymous>) at C:\Users\\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\onedrive-api.js:8:71"`

I  encountered a alike bug on mac, which said "OneDriveApi: Got unhandled error....", I guess it's a problem of one drive.

**Comment by kingzevin** [comment]
@cat-walk Is your language specified as Chinese? What if you change that to English? The "应用" token in the invalid url seems suspicious.

**Comment by LordHumphrey** [comment]
> @cat-walk Is your language specified as Chinese? What if you change that to English? The "应用" token in the invalid url seems suspicious.

But this folder is created by joplin?So I must change the language of OneDrive?

**Comment by kingzevin** [comment]
You may try to change the system language and I suspect it works. Good luck. 
In addition, if you're using onedrive, then you can choose to synchronize the joplin data in the local file system, with the path specified as the onedrive sync folder.

**Comment by LordHumphrey** [comment]
> You may try to change the system language and I suspect it works. Good luck.
> In addition, if you're using onedrive, then you can choose to synchronize the joplin data in the local file system, with the path specified as the onedrive sync folder. It's a workaround.

Thank you.I will try later.But I wish joplin fix this bug,especially as a global opensource software. There may be some grammatical errors in this sentence, I'm sorry.Thank you!

**Comment by kingzevin** [comment]
That's fine. I hope they can fix it. Good luck!

**Comment by krisnyse** [comment]
Same problem here. Unable to sync Joplin on my Android phone.
"Last error: Error: Not a valid URL"

**Comment by willemswang** [comment]
Same problem here. Unable to sync Joplin on my iphone and MAC.
Completed: 2021.04.18 15:47
Last error: Error: Not a valid URL: https://graph.microsoft.com/v1.0/drives/b!TQrdju4Q-UCnxYZUQ4EkXOGNI9NK42lHu_nnAkJZHRV6gZ7G_M4tTY5TfFUT4dVm/root:/应用/Joplin/info.json:/content

**Comment by qrw** [comment]
Same problem and want to fix it

**Comment by AyFun** [comment]
Same problem here.

**Comment by emgrey02** [comment]
I have an error where i have an Apps/Joplin folder on my OneDrive with all the notes I want to sync, but when I try to sync it fails because it is trying to read Apps/Joplin 1, which it creates when I synchronize. If it helps I have a windows computer and a linux computer and I want to sync joplin notes between the two.

**Comment by zzxCNCZ** [comment]
These steps work for me:
1. backup your ondrive joplin files
2. joplin load files(backup files) from your disk by file system
3. delete ondrive joplin folder
4. change joplin app sync type to ondrive
5. re-upload local file to ondrive

Then it works. After that ,your other joplin client should re sync from ondrive.

**Comment by rileyhawk1417** [comment]
OneDrive authenticates for me but does not grab the files from OneDrive.
On android its working fine but not sure why its not working on the latest release for desktop.
Not sure if the API for onedrive is broken, but can't retrieve my notes.
The log below happens in the beginning of the sync process.

```txt
2022-02-01 14:40:21: OneDriveApi: "Got unhandled error:", "ERR_TLS_CERT_ALTNAME_INVALID", "request to https://public.db.files.1drv.com/y4mKJzWBPkF1-C6ebBWfaV2FNXEkyWsYpxAfvUER1lLfK8dq6Dom-YTC7atQ2LDUfp8wYY7RnrEgu60Rui3Ymj2kSyKV4TuGJypl_8asU92dcZJv8zkuwSdWT3UlIcYVkUhG5X85PbsKOYG1BvW3JoP5FUdJFSbyLZACk03g2t6wXvYDFlSGeUYQJNB2a1i1XvBl-rgRnH6qp0G82FWSmWguretpomOwjdrKLeCiNqN87g/0a33e8bc9ef34740bcf4e0fdb9b8cdf6.md failed, reason: Hostname/IP does not match certificate's altnames: Host: public.db.files.1drv.com. is not in the cert's altnames: DNS:*.fp.measure.office.com, DNS:graph.microsoft.com", "FetchError: request to https://public.db.files.1drv.com/y4mKJzWBPkF1-C6ebBWfaV2FNXEkyWsYpxAfvUER1lLfK8dq6Dom-YTC7atQ2LDUfp8wYY7RnrEgu60Rui3Ymj2kSyKV4TuGJypl_8asU92dcZJv8zkuwSdWT3UlIcYVkUhG5X85PbsKOYG1BvW3JoP5FUdJFSbyLZACk03g2t6wXvYDFlSGeUYQJNB2a1i1XvBl-rgRnH6qp0G82FWSmWguretpomOwjdrKLeCiNqN87g/0a33e8bc9ef34740bcf4e0fdb9b8cdf6.md failed, reason: Hostname/IP does not match certificate's altnames: Host: public.db.files.1drv.com. is not in the cert's altnames: DNS:*.fp.measure.office.com, DNS:graph.microsoft.com
Code: ERR_TLS_CERT_ALTNAME_INVALID
FetchError: request to https://public.db.files.1drv.com/y4mKJzWBPkF1-C6ebBWfaV2FNXEkyWsYpxAfvUER1lLfK8dq6Dom-YTC7atQ2LDUfp8wYY7RnrEgu60Rui3Ymj2kSyKV4TuGJypl_8asU92dcZJv8zkuwSdWT3UlIcYVkUhG5X85PbsKOYG1BvW3JoP5FUdJFSbyLZACk03g2t6wXvYDFlSGeUYQJNB2a1i1XvBl-rgRnH6qp0G82FWSmWguretpomOwjdrKLeCiNqN87g/0a33e8bc9ef34740bcf4e0fdb9b8cdf6.md failed, reason: Hostname/IP does not match certificate's altnames: Host: public.db.files.1drv.com. is not in the cert's altnames: DNS:*.fp.measure.office.com, DNS:graph.microsoft.com
    at ClientRequest.<anonymous> (/tmp/.mount_Joplin0exMf8/resources/app.asar/node_modules/@joplin/lib/node_modules/node-fetch/index.js:133:11)
    at ClientRequest.emit (events.js:376:20)
    at TLSSocket.socketErrorListener (_http_client.js:475:9)
    at TLSSocket.emit (events.js:376:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)"
```

**Comment by HuaWang135608** [comment]
确保网络连接正常
退出 Joplin（Ctrl + Q）重新打开就好了
Ensuring that the network connection is normal.
Quitting Joplin (Ctrl + Q) and just open it again.

**Comment by wh201906** [comment]
I think the `Not a valid URL` bug is fixed in #6916
@laurent22 Hi. Do you think it should be closed?

**Comment by laurent22** [comment]
That's right, I think it's fixed now so let's close
